models/QtiRunnerInitDataBuilder.php

```diff
@@ -228,21 +229,25 @@ private function fillItemsData($itemId, $itemRef, $itemData)
         ];
     }
 
-    protected function getResultVariables($resultId, $filterSubmission, $filterTypes = array())
+    /**
+     * @throws common_exception_Error
+     * @throws common_Exception
+     */
+    protected function getResultVariables($resultId, $filterSubmission, $filterTypes = []): array
```

Comment:
That would be a great moment to create a test for this class, at least to cover these changes to ensure we won't have the same issue again.